test/connectSpec.js

```diff
@@ -110,6 +110,37 @@ describe('Connection methods', function () {
       infoNock.done()
     })
 
+    it('should set urls from object in metadata', function * () {
```

Comment:
Should also test that no unexpected extra properties make it into `plugin.urls`.